src/metabase/api/database.clj

```diff
@@ -1206,6 +1206,14 @@
                                      [:in :collection_id (api/check-404 (not-empty (t2/select-pks-set :model/Collection :name schema)))])])
          (map api.table/card->virtual-table))))
 
+(api.macros/defendpoint :get "/:id/healthcheck"
+  "Reports whether the database can currently connect"
+  [{:keys [id]} :- [:map [:id ms/PositiveInt]]]
+  (let [{:keys [engine details]} (t2/select-one :model/Database :id id)]
+    (if-let [err-map (test-database-connection engine details)]
```

Comment:
is this function running some kind of query on the database or seeing if a connection can be started?